packages/plugin-interactive-tools/sources/commands/search.tsx

```diff
@@ -0,0 +1,239 @@
+import {BaseCommand}                         from '@yarnpkg/cli';
+import {Configuration, FormatType}           from '@yarnpkg/core';
+import {ScrollableItems}                     from '@yarnpkg/libui/sources/components/ScrollableItems';
+import {useMinistore}                        from '@yarnpkg/libui/sources/hooks/useMinistore';
+import {useSpace}                            from '@yarnpkg/libui/sources/hooks/useSpace';
+import {renderForm, SubmitInjectedComponent} from '@yarnpkg/libui/sources/misc/renderForm';
+import {Command, Usage}                      from 'clipanion';
+import InkTextInput, {InkTextInputProps}     from 'ink-text-input';
+import {Box, Text, Color}                    from 'ink';
+import React, {useEffect, useState}          from 'react';
+
+import {AlgoliaPackage, search}              from '../algolia';
+
+const targets = [`regular`, `dev`, `peer`];
+
+// eslint-disable-next-line arca/no-default-export
+export default class SearchCommand extends BaseCommand {
+  static usage: Usage = Command.Usage({
+    category: `Interactive commands`,
+    description: `open the search interface`,
+    details: `
+    This command opens a fullscreen terminal interface where you can search for and install packages from the npm registry.
+    `,
+    examples: [[
+      `Open the search window`,
+      `yarn search`,
+    ]],
+  });
+
+  @Command.Path(`search`)
+  async execute() {
+    const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
+
+    const prettyName = (name: string) => {
+      const hasScope = name.includes(`/`);
+
+      if (hasScope) {
+        const [scope, descriptor] = name.split(`/`);
+        return <>{configuration.format(`${scope}/`, FormatType.SCOPE)}{configuration.format(descriptor, FormatType.NAME)}</>;
+      } else {
+        return configuration.format(name, FormatType.NAME);
+      }
+    };
+
+    const Prompt = () => {
+      return (
+        <Box flexDirection="row">
+          <Box flexDirection="column" width={49}>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<up>`}</Color>/<Color bold cyanBright>{`<down>`}</Color> to move between packages.
+            </Box>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<space>`}</Color> to select a package.
+            </Box>
+            <Box marginLeft={1}>
+              Press <Color bold cyanBright>{`<space>`}</Color> again to change the target.
+            </Box>
+          </Box>
+          <Box flexDirection="column">
+            <Box marginLeft={1}>
+              Press <Color bold cyanBright>{`<enter>`}</Color> to install the selected packages.
+            </Box>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<ctrl+c>`}</Color> to abort.
+            </Box>
+          </Box>
+        </Box>
```

Comment:
Doesn't bother me personally, but be prepared for @arcanis to bring 2 x 2 symmetry in the discussion. 😄

Reply:
😬

Reply:
My symmetry hurts 😢 But in this case there really isn't another place to put it, so I guess it'll be fine, ahah 😄